nanofi/examples/tail_file.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+    *
+    *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#include "api/nanofi.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <limits.h>
+#include <sys/stat.h>
+
+typedef struct flow_file_records {
+    flow_file_record ** records;
+    uint64_t len;
+} flow_file_records;
+
+typedef struct file_offset {
+    int offset;
+} file_offset;
+
+struct flow_file_records * flowfiles = NULL;
+struct file_offset * fileoffset = NULL;
+struct nifi_instance * instance = NULL;
+
+void transmit_flow_files(nifi_instance * instance) {
+    if (flowfiles) {
+        for (int i = 0; i < flowfiles->len; ++i) {
+            if (flowfiles->records[i]) {
+                transmit_flowfile(flowfiles->records[i], instance);
+            }
+        }
+    }
+}
+
+char * get_string(char * begin, char * end) {
+
+    if (begin == NULL || (end != NULL && end < begin)) {
+        return NULL;
+    }
+
+    int len = 0;
+    char * buff = NULL;
+
+    if (begin == end) {
+        buff = (char *)malloc(2 * sizeof(char));
+        strncpy(buff, begin, 1);
+        buff[1] = '\0';
+        return buff;
+    }
+
+    if (end) {
```

Comment:
In these two blocks only the first line is different.

```
        len = end ? (end-begin) : strlen(begin);
        buff = (char *)malloc((len+1) * sizeof(char));
        strncpy(buff, begin, len);
        buff[len] = '\0';
```

Reply:
This function is no more present in the latest update.